### DISCUSSION TITLE
docTR half precision - accuracy and speed benchmark

### DISCUSSION THREAD

**Discussion by tonywu71** [mindee/doctr#1385] [Q&A]
# Inquiry: docTR half-precision - accuracy and speed benchmark

Hello,

As a `docTR` enthusiast, I've been exploring the `docTR` model recently and I'm curious if anyone has conducted benchmarks using half-precision (16-bit floating-point format) to assess its performance in terms of accuracy and speed. See the [documentation](https://mindee.github.io/doctr/latest/using_doctr/using_model_export.html#half-precision) for reference.

As of today, I believe one bottleneck of `docTR` is the cost of running the model on GPU compared to CPU usage. Thus, I think knowing the amount of time saved using half-precision could be extremely useful to the docTR community. 😁

Looking forward to your experiences and discussions!

Regards,
Tony

**Comment by felixT2K** [comment]
Hi @tonywu71 👋🏼,

That's a really great idea 🤗 
But personally i would wait for it until #1364 is done :)

For the moment i can share some information from testing it (especially pytorch):

- vitstr_base -> does not work atm with full fp16 (maybe with amp i haven't tested it yet)
- crnn_vgg16_bn -> high accuracy drop
- vitstr_small / master / parseq / crnn_mobilenet_v3_small / crnn_mobilenet_v3_large -> on a first view no acc drop 
- sar (raises an exception) / but sar is still a bit broken in both frameworks we will have an look after the next release :)

About the whole OCR pipeline as mentioned i haven't benchmarked it yet but from feeling it is much much faster (RTX3080 mobile)

Best regards,
Felix

**Reply by tonywu71** [reply]
Thanks so much @felixT2K 🙏🏼
